src/routes/safe/components/Apps/hooks/useIframeMessageHandler.ts

```diff
@@ -70,17 +71,31 @@ const useIframeMessageHandler = (
   )
 
   useEffect(() => {
-    const handleIframeMessage = (msg: CustomMessageEvent) => {
-      if (!msg?.data.messageId) {
+    const handleIframeMessage = (
+      messageId: SDKMessageIds,
+      messagePayload: SDKMessageToPayload[typeof messageId],
+      requestId: RequestId,
+    ) => {
+      if (!messageId) {
         console.error('ThirdPartyApp: A message was received without message id.')
         return
       }
-      const { requestId } = msg.data
 
-      switch (msg.data.messageId) {
+      switch (messageId) {
+        // typescript doesn't narrow type in switch/case statements
+        // issue: https://github.com/microsoft/TypeScript/issues/20375
+        // possible solution: https://stackoverflow.com/a/43879897/7820085
         case SDK_MESSAGES.SEND_TRANSACTIONS: {
-          if (msg.data.data) {
-            openConfirmationModal(msg.data.data, requestId)
+          if (messagePayload) {
+            openConfirmationModal(messagePayload as SDKMessageToPayload['SEND_TRANSACTIONS'], undefined, requestId)
```

Comment:
Instead of the string `SEND_TRANSACTIONS` wouldit make sense to use `SDK_MESSAGES.SEND_TRANSACTIONS` ... or for this case we should be able to use `messageId` (as we check it with the `case`)

Reply:
@rmeissner we won't be able to use `messageId`, there's a comment few lines above, maybe i'll address this in one of the future PRs :)
```
        // typescript doesn't narrow type in switch/case statements
        // issue: https://github.com/microsoft/TypeScript/issues/20375
        // possible solution: https://stackoverflow.com/a/43879897/7820085
```

Reply:
I read the TS issue ... thougt this was mainly that the type is not automatically narrowed down and that you have to cast. Still feels weird that we have to use the string ... but so be it :P